modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java

```diff
@@ -313,6 +319,110 @@ public void testPutObjectDefinition() throws Exception {
 		_objectDefinitionLocalService.deleteObjectDefinition(
 			postObjectDefinition.getId());
 
+		// Modifiable System Object
+
+		ObjectDefinition randomModifiableSystemObjectDefinition =
+			_addObjectDefinition(_randomModifiableSystemObjectDefinition());
+
+		randomModifiableSystemObjectDefinition.setEnableObjectEntryDraft(
+			(Boolean)null);
+		randomModifiableSystemObjectDefinition.
+			setObjectFolderExternalReferenceCode("");
+
+		ObjectValidationRule systemObjectValidationRule =
+			(ObjectValidationRule)ArrayUtil.getValue(
+				randomModifiableSystemObjectDefinition.
+					getObjectValidationRules(),
+				1);
+
+		ObjectValidationRuleSetting[] updatedObjectValidationRuleSettings = {
+			new ObjectValidationRuleSetting() {
+				{
+					name =
+						ObjectValidationRuleSettingConstants.
+							NAME_OUTPUT_OBJECT_FIELD_EXTERNAL_REFERENCE_CODE;
+					value = "customObjectFieldERC";
+				}
+			}
+		};
+
+		ObjectValidationRule updatedCustomObjectValidationRule =
+			new ObjectValidationRule() {
+				{
+					active = false;
+					engine = ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY;
+					errorLabel = Collections.singletonMap(
+						"en_US", RandomTestUtil.randomString());
+					externalReferenceCode = "customObjectValidationRuleERC";
+					name = Collections.singletonMap(
+						"en_US", RandomTestUtil.randomString());
+					objectDefinitionExternalReferenceCode =
+						randomModifiableSystemObjectDefinition.
+							getExternalReferenceCode();
+					objectValidationRuleSettings =
+						updatedObjectValidationRuleSettings;
+					outputType = OutputType.create("partialValidation");
+					script = RandomTestUtil.randomString();
+					system = false;
+				}
+			};
+
+		ObjectValidationRule updatedSystemObjectValidationRule =
+			new ObjectValidationRule() {
+				{
+					active = false;
+					engine = ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY;
+					errorLabel = Collections.singletonMap(
+						"en_US", RandomTestUtil.randomString());
+					externalReferenceCode = "systemObjectValidationRuleERC";
+					name = Collections.singletonMap(
+						"en_US", RandomTestUtil.randomString());
+					objectDefinitionExternalReferenceCode =
+						randomModifiableSystemObjectDefinition.
+							getExternalReferenceCode();
+					objectValidationRuleSettings =
+						updatedObjectValidationRuleSettings;
+					outputType = OutputType.create("partialValidation");
+					script = RandomTestUtil.randomString();
+					system = true;
+				}
+			};
+
+		randomModifiableSystemObjectDefinition.setObjectValidationRules(
+			new ObjectValidationRule[] {
+				updatedCustomObjectValidationRule,
+				updatedSystemObjectValidationRule
+			});
```

Comment:
@pedro-oliveira446 please inline the `updatedSystemObjectValidationRule` since we are not using it anywhere

```
randomModifiableSystemObjectDefinition.setObjectValidationRules(
	new ObjectValidationRule[] {
		updatedCustomObjectValidationRule,
		new ObjectValidationRule() {
			{
				active = false;
				engine =
					ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY;
				errorLabel = Collections.singletonMap(
					"en_US", RandomTestUtil.randomString());
				externalReferenceCode = "systemObjectValidationRuleERC";
				name = Collections.singletonMap(
					"en_US", RandomTestUtil.randomString());
				objectDefinitionExternalReferenceCode =
					randomModifiableSystemObjectDefinition.
						getExternalReferenceCode();
				objectValidationRuleSettings =
					new ObjectValidationRuleSetting[] {
						new ObjectValidationRuleSetting() {
							{
								name =
									ObjectValidationRuleSettingConstants.NAME_OUTPUT_OBJECT_FIELD_EXTERNAL_REFERENCE_CODE;
								value = "customObjectFieldERC";
							}
						}
					};
				outputType = OutputType.create("partialValidation");
				script = RandomTestUtil.randomString();
				system = true;
			}
		}
	});
```